cli/src/main/java/org/jboss/pnc/bacon/cli/App.java

```diff
@@ -258,11 +262,15 @@ private void init(String command) {
         String endpoint = System.getenv("OTEL_EXPORTER_OTLP_ENDPOINT");
         String service = System.getenv("OTEL_SERVICE_NAME");
         if (endpoint != null) {
-            log.debug("Enabling OTEL collection on {} with service name {}", endpoint, service);
-            OTELHelper.startOtel(
-                    service,
-                    command,
-                    OtelCLIHelper.defaultSpanProcessor(OtelCLIHelper.defaultSpanExporter(endpoint)));
+            log.debug("Enabling OpenTelemetry collection on {} with service name {}", endpoint, service);
+            OTelCLIHelper.startOTel(
+                service,
+                command,
+                OTelCLIHelper.defaultSpanProcessor( OTelCLIHelper.defaultSpanExporter( endpoint)));
+            SpanContext current = Span.current().getSpanContext();
+            MDCUtils.addMDCFromOtelHeadersWithFallback( null, current, true );
+            // Temporary until MdcToHeadersFilter also handles the traceparent header.
+            OtelUtils.createTraceParentHeader( current ).forEach( MDC::put );
```

Comment:
When https://github.com/project-ncl/pnc/pull/3654/ is merged this can be removed